pipeline/commands/upright_transform.py

```diff
@@ -0,0 +1,146 @@
+import os
+import json
+import xarray as xr
+
+from typing import Dict, Any, Union, Optional
+
+import boto3
+
+import numpy as np
+from neuron_morphology.swc_io import morphology_from_swc
+import neuron_morphology.transforms.affine_transform as aff
+from neuron_morphology.transforms.upright_angle.compute_angle import get_upright_angle, calculate_transform
+from neuron_morphology.swc_io import morphology_to_swc
+from neuron_morphology.morphology import Morphology
+
+from harness import step_fns_ecs_harness
+
+
+s3 = boto3.client("s3")
+
+
+def collect_inputs(
+        working_bucket: str, 
+        run_prefix: str, 
+        morphology_scaled_key: str,
+        gradient_field_key: str
+ ) -> Dict[str, Any]:
+    """
+    Gather from AWS the inputs required to run the depth field module.
+
+    Parameters
+    ----------
+    working_bucket : The name of this pipeline's working bucket
+    run_prefix : This run's resources (within the working bucket) have keys 
+        beginning with this prefix.
+    morphology_scaled_key : identifier for the scaled corrected morphology
+    gradient_field_key: identifier for the gradient field
+
+    Returns
+    -------
+    morphology_data : Morphology object of scale corrected reconstruction
+    gradient_field_data : xarray object of the gradient field
+    """
+
+    # boto3 get bytes from s3 working buckets
+    gradient_field_obj = s3.get_object(Bucket=working_bucket, Key=gradient_field_key)
```

Comment:
For clarity you could suffix this "response" vs. "obj"

Reply:
changed